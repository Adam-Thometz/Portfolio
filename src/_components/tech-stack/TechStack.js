import React from "react";
import techStackInfo from "./techStackInfo";
import TechStackCard from "./tech-stack-card/TechStackCard";
import './TechStack.css'

const TechStack = ({ techStack }) => {
  return (
    <div className="TechStack">
      <h3>Tech Stack Used</h3>
      <div className="TechStack-main">
        {techStack.map(tech => {
          const t = techStackInfo.find(currTech => currTech.name === tech)
          return <TechStackCard name={t.name} icon={t.icon} />
        })}
      </div>
    </div>
  );
};

export default TechStack;