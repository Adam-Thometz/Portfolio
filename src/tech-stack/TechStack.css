.TechStack {
  display: flex;
  flex-direction: column;
}