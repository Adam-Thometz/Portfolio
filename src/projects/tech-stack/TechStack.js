import React from "react";
import techStackInfo from "../../_data/techStackInfo";
import TechStackCard from "./tech-stack-card/TechStackCard";
import './TechStack.css'

const TechStack = ({ techStack }) => {
  return (
    <div className="TechStack">
      <h3>Tech Stack Used</h3>
      <div className="TechStack-main">
        {techStack.map(tech => {
          const { name, icon } = techStackInfo.find(currTech => currTech.name === tech)
          return <TechStackCard name={name} icon={icon} />
        })}
      </div>
    </div>
  );
};

export default TechStack;