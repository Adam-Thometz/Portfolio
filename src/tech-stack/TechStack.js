import React from "react";
import techStackInfo from "./techStackInfo";
import TechStackCard from "./tech-stack-card/TechStackCard";
import "./TechStack.css";

const TechStack = ({ techStack }) => {
  return (
    <section className="TechStack">
      {techStack.map((tech) => {
        const { name, icon } = techStackInfo.find(
          (currTech) => currTech.name === tech
        );
        return <TechStackCard name={name} icon={icon} />;
      })}
    </section>
  );
};

export default TechStack;
