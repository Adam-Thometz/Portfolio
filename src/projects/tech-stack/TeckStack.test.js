import React from "react";
import { render } from "@testing-library/react";
import TechStack from './TechStack';

describe('TechStack component', () => {
  it('renders without crashing', () => {
    render(<TechStack techStack={["Javascript"]} />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<TechStack techStack={["Javascript"]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});