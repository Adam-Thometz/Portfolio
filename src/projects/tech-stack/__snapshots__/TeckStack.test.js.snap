// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechStack component matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="TechStack"
  >
    <figure
      class="TechStackCard"
    >
      <img
        alt="Javascript"
        class="TechStackCard-img"
        src="javascript.png"
      />
      <figcaption
        class="TechStackCard-caption"
      >
        Javascript
      </figcaption>
    </figure>
  </section>
</DocumentFragment>
`;
